rideshare.rb

```diff
@@ -0,0 +1,151 @@
+# RIDESHARE
+# Calculates statistics of rideshare data.
+
+drivers = {
+  DR0001: [
+    {
+      "3rd Feb 2016": [
+        {
+          cost: 10,
+          rider_id: "RD0003",
+          rating: 3,
+        },
+        {
+          cost: 30,
+          rider_id: "RD0015",
+          rating: 4,
+        },
+      ],
+    },
+    {
+      "5th Feb 2016": [
+        {
+          cost: 45,
+          rider_id: "RD0003",
+          rating: 2,
+        },
+      ],
+    },
+  ],
+  DR0002: [
+    {
+      "3rd Feb 2016": [
+        {
+          cost: 25,
+          rider_id: "RD0073",
+          rating: 5,
+        },
+      ],
+    },
+    {
+      "4th Feb 2016": [
+        {
+          cost: 15,
+          rider_id: "RD0013",
+          rating: 1,
+        },
+      ],
+    },
+    {
+      "5th Feb 2016": [
+        {
+          cost: 35,
+          rider_id: "RD0066",
+          rating: 3,
+        },
+      ],
+    },
+  ],
+  DR0003: [
+    {
+      "4th Feb 2016": [
+        {
+          cost: 5,
+          rider_id: "RD0066",
+          rating: 5,
+        },
+      ],
+    },
+    {
+      "5th Feb 2016": [
+        {
+          cost: 50,
+          rider_id: "RD0003",
+          rating: 2,
+        },
+      ],
+    },
+  ],
+  DR0004: [
+    {
+      "3rd Feb 2016": [
+        {
+          cost: 5,
+          rider_id: "RD0022",
+          rating: 5
+        },
+      ],
+    },
+    {
+      "4th Feb 2016": [
+        {
+          cost: 10,
+          rider_id: "RD0022",
+          rating: 4
+        },
+      ],
+    },
+    {
+      "5th Feb 2016": [
+        {
+          cost: 20,
+          rider_id: "RD0073",
+          rating: 5,
+        },
+      ],
+    },
+  ],
+}
+
+highest_earner = {
+  driver: "",
+  amount_earned: 0,
+}
+
+highest_rated = {
+  driver: "",
+  rating: 0,
+}
+
+puts "STATS PER DRIVER\n**********\n\n"
+drivers.each do | driver, workdays |
+  puts "#{driver} stats:"
+
+  # The number of rides each driver has given
+  ride_count = workdays.sum { |dates, rides| dates.sum { | date, rides | rides.count } }
```

Comment:
Nice use of a block with `.sum`!  This lets you skip using a `.map`.

This is equivalent to doing:
```ruby
   ride_count = workdays.map { |dates, rides| dates.sum { | date, rides | rides.count } }.sum
```